src/OrleansServiceBus/Providers/Streams/EventHub/EventHubPartitionCheckpoint.cs

```diff
@@ -0,0 +1,81 @@
+﻿
+using System;
+using System.Threading.Tasks;
+using Microsoft.ServiceBus.Messaging;
+using Orleans.AzureUtils;
+
+namespace Orleans.ServiceBus.Providers
+{
+    /// <summary>
+    /// This class stores EventHub partition checkpoint information (a partition offset) in azure table storage.
+    /// TODO: Use dependency injection to fill this behavior.
+    /// This class stores data into table storage, which will not be ideal for all users.  Further, it introduces
+    ///   a dependency on azure storage to this assembly.  Ideally this should be an injectable behavior, rather
+    ///   than a concrete class.
+    /// </summary>
+    internal class EventHubPartitionCheckpoint
+    {
+        private readonly AzureTableDataManager<EventHubPartitionCheckpointEntity> dataManager;
+        private readonly TimeSpan persistInterval;
+
+        private EventHubPartitionCheckpointEntity entity;
+        private Task inProgressSave;
+        private DateTime? throttleSavesUntilUtc;
+
+        public bool Exists { get { return entity != null && entity.Offset != EventHubConsumerGroup.StartOfStream; } }
+
+        public EventHubPartitionCheckpoint(ICheckpointSettings settings, string streamProviderName, string partition)
+        {
+            if (settings == null)
+            {
+                throw new ArgumentNullException("settings");
+            }
+            if (string.IsNullOrWhiteSpace(streamProviderName))
+            {
+                throw new ArgumentNullException("streamProviderName");
+            }
+            if (string.IsNullOrWhiteSpace(partition))
+            {
+                throw new ArgumentNullException("partition");
+            }
+            persistInterval = settings.PersistInterval;
+            dataManager = new AzureTableDataManager<EventHubPartitionCheckpointEntity>(settings.TableName, settings.DataConnectionString);
+            entity = EventHubPartitionCheckpointEntity.Create(streamProviderName, settings.ServiceId, partition);
+        }
+
+        public Task Initialize()
+        {
+            return dataManager.InitTableAsync();
```

Comment:
Instead of having a public constructor + Initialize method to build this object, did you consider just having 
`public static Task<EventHubPartitionCheckpoint> Create(ICheckpointSettings settings, string streamProviderName, string partition)`
and making the constructor private? I tend to avoid having references to objects that weren't fully initialized. Not sure if it fits this scenario as it might make it harder to inject later on, and I know we have a lot of Initialize methods all over the place, but no harm in asking